### PR TITLE
render sensible labels for nodes with little/no metadata

### PR DESCRIPTION
**Pull request by rade**
A report may contain references to nodes which are not contained in the report. For example a short-lived process may have initiated some connections, which Scope detected, but died before Scope got round to obtaining full details for inclusion in the process topology.

More generally, we may be missing some or all metadata for a node - having just a reference (which is the node ID) is the most extreme case.

We encounter this missing info when summarising nodes for rendering in the UI (i.e. determining their shape, labels, etc), and when rendering links to them as parents, children and connection sources/destinations in the detail panel.

This PR ensures that we can summarise nodes in all these cases, ensuring that we can cope with having nothing more than the ID and topology.

Note that there are some cases where our renderers throw away nodes that miss some metadata, when really they should be passing them through. Fixing that entails small changes scattered across a bunch of files, which I'd rather defer to a separate PR.

### CODE REVIEW

File: render/detailed/connections.go [hunk 1]
```diff
@@ -125,10 +125,8 @@ func internetAddr(node report.Node, ep report.Node) (string, bool) {
 func (c *connectionCounters) rows(r report.Report, ns report.Nodes, includeLocal bool) []Connection {
 	output := []Connection{}
 	for row, count := range c.counts {
-		// Use MakeNodeSummary to render the id and label of this node
-		// TODO(paulbellamy): Would be cleaner if we hade just a
-		// MakeNodeID(ns[row.remoteNodeID]). As we don't need the whole summary.
-		summary, _ := MakeNodeSummary(report.RenderContext{Report: r}, ns[row.remoteNodeID])
+		// Use MakeBasicNodeSummary to render the id and label of this node
+		summary, _ := MakeBasicNodeSummary(r, ns[row.remoteNodeID])
```
Comment (bboreham) [inline render/detailed/connections.go]: I had the impression that if Paul's comment were actioned then we could drop the `Report` parameter, which may improve matters upstream. But it's difficult to check on my phone.

Comment (rade) [reply render/detailed/connections.go]: We are very close to that. Very close indeed. I shall take a look after this PR.

File: render/detailed/summary.go [hunk 2]
```diff
@@ -132,140 +163,134 @@ func (n NodeSummary) SummarizeMetrics() NodeSummary {
...
-		base.Shape = report.Circle
-		return base, true
+	default:
+		// try rendering it as an endpoint
+		if _, addr, _, ok := report.ParseEndpointNodeID(n.ID); ok {
+			base.Label = addr
+			base.Shape = report.Circle
+		} else {
+			// last resort
+			base.Label = pseudoID
```
[the remark this answers is not in the document]
Comment (rade) [reply render/detailed/summary.go]: I've just added something to the top level logic to set the default shape to a triangle - a kind of warning that we don't really know what this thing is :)

File: render/detailed/summary.go [hunk 3]
```diff
@@ -132,140 +163,134 @@ func (n NodeSummary) SummarizeMetrics() NodeSummary {
...
-		base.Label, _ = n.Latest.Lookup(docker.ImageID)
-		if len(base.Label) > 12 {
-			base.Label = base.Label[:12]
-		}
+func containerImageNodeSummary(base BasicNodeSummary, n report.Node) BasicNodeSummary {
+	var (
+		imageName, _            = n.Latest.Lookup(docker.ImageName)
+		imageNameWithoutVersion = docker.ImageNameWithoutVersion(imageName)
+	)
+	switch true {
```
[the remark this answers is not in the document]
Comment (rade) [reply render/detailed/summary.go]: oops. Fixed here and elsewhere.

**Pull request merged**